BHoM/BHoM_Engine
# Diffing_Engine: allow to use Id stored in CustomData; fix Explode `modifiedPropertiesPerObj`

### Issues addressed by this PR


Closes #1665 
Closes #1660 



This PR primarily adds the feature where you can use an Id stored in CustomData instead of the object Hash, as explained in #1665.

Additionally, it also fixes the Explode not working properly for the `Diff` property called `modifiedPropertiesPerObj`, see #1660. It does so:
- by implementing a workaround to make the Explode work. However, going through the various exploded levels is not very user friendly. Therefore:
- it adds an Engine method `ListModifiedProperties` that "properly" queries and displays the data.


### Test files

@kayleighhoude this should enable the workflow in Revit. 

![image](https://user-images.githubusercontent.com/6352844/80465744-78213100-8933-11ea-9079-0a135f46b9c8.png)


Workflow:
- Create your model in Revit
- Pull it in Grasshopper. Internalize the objects somewhere.
- Go back to Revit. Modify the model.
- Pull it again in Grasshopper.
- Attach the internalized objects and the newly pulled model to the `Diffing` Engine method as per image above. You need `customdataIdName` to be set to ~~`Revit_elementId`~~ `Revit_uniqueId`.
- Check the diff:
  - explode the Diff
  - check what properties have been modified using the `ListModifiedProperties` method as in the picture above. 

### Changelog


- Can now input a CustomData Id Name to the Diffing method so it will be using a certain software id to recognise objects instead of hashes.
- Added `ListModifiedProperties` method to better view the Diff property `modifiedPropertiesPerObj`
- Added workaround to make BHoM Explode work properly with `modifiedPropertiesPerObj`
